tests/test_schema.py

```diff
@@ -1022,6 +1235,264 @@ def test_authentication(self):
+                                  variables=params)
+
+        # Check result
+        proj = executed['data']['addProject']['project']
+        self.assertEqual(proj['name'], 'Example')
+        self.assertEqual(proj['title'], 'Example title')
+        self.assertEqual(proj['parentProject'], None)
+
+        eco = proj['ecosystem']
+        self.assertEqual(eco['id'], str(self.eco.id))
+        self.assertEqual(eco['name'], 'Eco-example')
+
+        # Check database
+        proj_db = Project.objects.get(id=int(proj['id']))
+        self.assertEqual(proj_db.id, int(proj['id']))
+        self.assertEqual(proj_db.name, 'Example')
+        self.assertEqual(proj_db.title, 'Example title')
+        self.assertEqual(proj_db.parent_project, None)
+        self.assertEqual(proj_db.ecosystem, self.eco)
+
+    def test_ecosystem_empty(self):
+        """Check whether projects with empty ecosystem cannot be added"""
+
+        params = {
+            'name': '',
+            'title': 'Example title',
+            'ecosystemId': ''
+        }
+        client = graphene.test.Client(schema)
+        executed = client.execute(self.BT_ADD_PROJECT,
+                                  context_value=self.context_value,
+                                  variables=params)
+
+        # Check error
+        msg = executed['errors'][0]['message']
+        self.assertEqual(msg, ECOSYSTEM_ID_NONE_ERROR)
+
+        # Check database
+        projects = Project.objects.all()
+        self.assertEqual(len(projects), 0)
+
+    def test_ecosystem_not_exists(self):
+        """Check whether projects cannot be added when the ecosystem is not found"""
+
+        params = {
+            'name': 'Test example',
+            'title': 'Example title',
+            'ecosystemId': '11111111'
+        }
+        client = graphene.test.Client(schema)
+        executed = client.execute(self.BT_ADD_PROJECT,
+                                  context_value=self.context_value,
+                                  variables=params)
+
+        # Check error
+        msg = executed['errors'][0]['message']
+        self.assertEqual(msg, ECOSYSTEM_DOES_NOT_EXIST_ERROR)
+
+        # Check database
+        projects = Project.objects.all()
+        self.assertEqual(len(projects), 0)
+
+    def test_name_empty(self):
+        """Check whether projects with empty names cannot be added"""
+
+        params = {
+            'name': '',
+            'title': 'Example title',
+            'ecosystemId': self.eco.id
+        }
+        client = graphene.test.Client(schema)
+        executed = client.execute(self.BT_ADD_PROJECT,
+                                  context_value=self.context_value,
+                                  variables=params)
+
+        # Check error
+        msg = executed['errors'][0]['message']
+        self.assertEqual(msg, NAME_EMPTY_ERROR)
+
+        # Check database
+        projects = Project.objects.all()
+        self.assertEqual(len(projects), 0)
+
```

Comment:
I'm not sure what to do with this test and the others you defined below. It's great to have them but I'm not sure if it delays the process to create new functionalities. We are having kind of the same tests in three different layers testing more or less the same. Probably for `db` and `api` they make sense because they test different things but for these one  it doesn't mind. We can suppose they use the underlying modules where we already test this. What do you think?

Reply:
I agree, as I do think that testing these kind of cases are more critical in `db` and `api`, as here we are checking that the errors propagated as expected. I propose to define a minimum set of tests that these mutations must have and then check for specific cases needing extra tests. WDYT?

Reply:
Let's do that!

Reply:
IMO, the basic tests would be:

* The one testing the functionality of the mutation
* Test authentication

And some tests with critical errors, in this case:
* Test when the ecosystem to add to the project is not found.
* Integrity error, when trying to add a project with an existing name.

Reply:
Ok. Let's do that.